back/api/views.py

```diff
@@ -2,11 +2,28 @@
 from rest_framework.response import Response
 from rest_framework.decorators import api_view
 from rest_framework.views import APIView
-from .models import User
+from .models import *
 from rest_framework.exceptions import AuthenticationFailed
-from .serializers import UserSerializer
+from .serializers import *
 import jwt, datetime
 # Create your views here.
+def authenticated(function):
+    def wrapper(self, request, *args):
+        token = request.COOKIES.get('jwt')
+        if not token:
+            raise AuthenticationFailed('Unauthentication')
+        
+        try:
+            payload = jwt.decode(token, 'secret', algorithms=['HS256'])
+
+        except jwt.ExpiredSignatureError:
+            raise AuthenticationFailed('Unauthentication')
+
+        args = (payload['id'],)
+
+        return function(self, request, *args)
```

Comment:
[KISS](https://en.wikipedia.org/wiki/KISS_principle)

```suggestion
        return function(self, request, payload['id'])
```

Reply:
Indeed it is clearer this way